core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/upgrade/DistroXUpgradeService.java

```diff
@@ -109,4 +139,24 @@ private void verifyCMLicenseValidity(String userCrn) {
         JsonCMLicense license = clouderaManagerLicenseProvider.getLicense(userCrn);
         paywallAccessChecker.checkPaywallAccess(license, paywallUrl);
     }
+
+    private void turnOffReplaceVmsParamIfRuntimeUpgrade(UpgradeV4Response upgradeV4Response, Stack stack, ImageInfoV4Response targetImage) {
+        if (upgradeV4Response.isReplaceVms()) {
+            try {
+                Image currentImage = componentConfigProviderService.getImage(stack.getId());
+                CloudbreakImageCatalogV3 imageCatalog = imageCatalogProvider.getImageCatalogV3(currentImage.getImageCatalogUrl());
+                com.sequenceiq.cloudbreak.cloud.model.catalog.Image currentCatalogImage =
+                        imageProvider.getCurrentImageFromCatalog(currentImage.getImageId(), imageCatalog);
+                com.sequenceiq.cloudbreak.cloud.model.catalog.Image targetCatalogImage =
+                        imageProvider.getCurrentImageFromCatalog(targetImage.getImageId(), imageCatalog);
+                if (!lockedComponentChecker.isUpgradePermitted(
+                        currentCatalogImage, targetCatalogImage, imageFilterParamsFactory.getStackRelatedParcels(stack))) {
+                    LOGGER.info("ReplaceVms parameter has been overridden to false in case of distrox runtime upgrade.");
```

Comment:
Can you give more context in the log, like CRN or cluster name and what is the source and target image id?